src/routers/cart.py

```diff
@@ -15,6 +15,7 @@
 @router.post("/checkout", status_code=status.HTTP_200_OK)
 async def checkout_cart(
     cart_api: CartAPISchema,
+    # Question: Curious to understand the use case of receiving the config as a dependency here.
```

Comment:
About the question:
I thought a lot about how to implement the discount logic on my server, since it was one of the most important parts of the code, inside of a simplistic approach. 
The dependency idea came from a similar approach I've seen with the usage of database integrations, where we fetch the database session with the same logic (e.g. `db = Depends(get_db)`). These "local" dependencies allow endpoints to access these variables directly and manipulate them, which I thought it was a very good match for the current design.
Finally, I also used a hint given by the FastAPI documentation for caching those settings (hence the reason I used the decorator on get_config), presented [here](https://fastapi.tiangolo.com/advanced/settings/#the-main-app-file).